docs/support/tutorials/user-spack.md

```diff
@@ -0,0 +1,132 @@
+[maijam@puhti-login11 ~]$ module purge
+The following modules were not unloaded:
+  (Use "module --force purge" to unload all):
+
+  1) csc-tools
+[maijam@puhti-login11 ~]$ export USER_SPACK_ROOT=/projappl/project_2001234/spack-instance-1
+[maijam@puhti-login11 ~]$ module load spack/v0.18-user
+Found existing user spack installation at /projappl/project_2001234/spack-instance-1
+```
+
+### Example build
+
+```bash
+[maijam@puhti-login11 ~]$ spack spec -I --reuse kakoune
+Input spec
+--------------------------------
+ -   kakoune
+
+Concretized
+--------------------------------
+ -   kakoune@2021.11.08%gcc@11.3.0 arch=linux-rhel8-cascadelake
+[^]      ^ncurses@6.2%gcc@11.3.0~symlinks+termlib abi=none arch=linux-rhel8-cascadelake
+[^]          ^pkgconf@1.8.0%gcc@11.3.0 arch=linux-rhel8-cascadelake
+[maijam@puhti-login11 ~]$ spack install --reuse kakoune
+[+] /appl/spack/v018/install-tree/gcc-11.3.0/pkgconf-1.8.0-2hkydh
+[+] /appl/spack/v018/install-tree/gcc-11.3.0/ncurses-6.2-n4phtn
+==> Installing kakoune-2021.11.08-yh4nmfid4st4l7gsmuzqv43o6qt6ckrm
+==> No binary for kakoune-2021.11.08-yh4nmfid4st4l7gsmuzqv43o6qt6ckrm found: installing from source
+==> Using cached archive: /local_scratch/maijam/spack-build-project_2002567/source-cache/_source-cache/archive/aa/aa30889d9da11331a243a8f40fe4f6a8619321b19217debac8f565e06eddb5f4.tar.bz2
+==> No patches needed for kakoune
+==> kakoune: Executing phase: 'edit'
+==> kakoune: Executing phase: 'build'
+==> kakoune: Executing phase: 'install'
+==> kakoune: Successfully installed kakoune-2021.11.08-yh4nmfid4st4l7gsmuzqv43o6qt6ckrm
+  Fetch: 0.00s.  Build: 34.31s.  Total: 34.31s.
+[+] /projappl/project_2001234/spack-instance-1/install_tree/gcc-11.3.0/kakoune-2021.11.08-yh4nmf
+```
+
+!!! info "Notes on commands"
+    The `spack spec` command shows what would be installed given a certain
+    input. The `-I` flag is used to display the current install status of the
+    package and its dependencies, while the `--reuse` flag will try to reuse
+    already installed dependencies whenever possible. Actual installation is
+    performed via the `spack install` command. Run `spack spec --help` and
+    `spack install --help` for further details.
+
+!!! info "Spec syntax"
+    The string that specifies which package should be installed (the *spec*) can
+    be simply just the package name, as above, but often you might want to install
+    a specific version, perhaps using a specific compiler and with some optional
+    installation flags (e.g. a GPU-enabled version of the software). Spack uses
+    a special syntax for specifying this information as explained in the [official
+    documentation](https://spack.readthedocs.io/en/latest/basic_usage.html#specs-dependencies).
+
+## Using modules with user Spack installations
+
+By default, Spack creates module files under `$USER_SPACK_ROOT/modules`
+and you can add that path to your `$MODULEPATH` with the command:
+
+```bash
+module use ${USER_SPACK_ROOT}/modules
```

Comment:
User spack module does not seem to create modulefiles by default. At least I did not get one following this example, resulting in an error upon running `module load kakoune` despite `module use ${USER_SPACK_ROOT}/modules`